freshmaker/handlers/botas/botas_shipped_advisory.py

```diff
@@ -149,3 +158,39 @@ def _filter_bundles_by_pinned_related_images(self, bundle_image_nvrs):
                     continue
                 ret_bundle_images_nvrs.add(nvr)
         return ret_bundle_images_nvrs
+
+    def get_published_original_nvr(self, session, rebuilt_nvr):
+        """
+        Search for an original build, that has been built and published to a
+            repository, and get original_nvr from it
+
+        :param session: database session
+        :param str rebuilt_nvr: rebuilt NVR to look build by
+        :rtype: str or None
+        :return: original NVR from the first published FM build for given NVR
+        """
+        original_nvr = None
+        # build event should be only one in database, or raise an error
+        build_event = db.session.query(ArtifactBuild).filter(
+            ArtifactBuild.rebuilt_nvr == rebuilt_nvr,
+            ArtifactBuild.type == ArtifactType.IMAGE.value,
+        ).one_or_none()
+        # recursively search for original build event
+        if build_event is not None:
+            original_nvr = build_event.original_nvr
+
+            # check if image is published
+            request_params = {'include': 'data.repositories'}
+            images = self._pyxis._pagination(f'images/nvr/{original_nvr}',
+                                             request_params)
+            if not images:
+                return None
+            # stop recursion if the image is published in some repo
+            if any([repo['published'] for repo in images[0].get('repositories')]):
```

Comment:
This is a minor optimization that allows `any` to stop processing the generator as soon as it matches a repo that is published.
```suggestion
            if any(repo['published'] for repo in images[0].get('repositories')):
```